app/jobs/single_submission_job.rb

```diff
@@ -0,0 +1,31 @@
+class SingleSubmissionJob < ActiveJob::Base
+  queue_as :submissions
+
+  # after_perform to refresh page? Race condition?
+
+  def apply_penalty_or_add_grace_credits(grouping,
```

Comment:
Let's make this a method of the `Submission` model, to avoid the code duplication.